functable.c

```diff
@@ -8,447 +8,476 @@
 # endif
-        functable.chunkmemset_safe = &chunkmemset_safe_sse2;
+        ft.chunkmemset_safe = &chunkmemset_safe_sse2;
 #endif
 #if defined(X86_SSE41) && defined(X86_SSE2)
     if (x86_cpu_has_sse41)
-        functable.chunkmemset_safe = &chunkmemset_safe_sse41;
+        ft.chunkmemset_safe = &chunkmemset_safe_sse41;
 #endif
 #ifdef X86_AVX_CHUNKSET
     if (x86_cpu_has_avx2)
-        functable.chunkmemset_safe = &chunkmemset_safe_avx;
+        ft.chunkmemset_safe = &chunkmemset_safe_avx;
 #endif
 #ifdef ARM_NEON_CHUNKSET
     if (arm_cpu_has_neon)
-        functable.chunkmemset_safe = &chunkmemset_safe_neon;
+        ft.chunkmemset_safe = &chunkmemset_safe_neon;
 #endif
 #ifdef POWER8_VSX_CHUNKSET
     if (power_cpu_has_arch_2_07)
-        functable.chunkmemset_safe = &chunkmemset_safe_power8;
+        ft.chunkmemset_safe = &chunkmemset_safe_power8;
 #endif
 
-    return functable.chunkmemset_safe(out, dist, len, left);
-}
-
-Z_INTERNAL uint32_t crc32_stub(uint32_t crc, const uint8_t *buf, size_t len) {
-    functable.crc32 = &PREFIX(crc32_braid);
-    cpu_check_features();
+    // crc32_stub:
+    ft.crc32 = &PREFIX(crc32_braid);
 #ifdef ARM_ACLE_CRC_HASH
     if (arm_cpu_has_crc32)
-        functable.crc32 = &crc32_acle;
+        ft.crc32 = &crc32_acle;
 #elif defined(POWER8_VSX_CRC32)
     if (power_cpu_has_arch_2_07)
-        functable.crc32 = &crc32_power8;
+        ft.crc32 = &crc32_power8;
 #elif defined(S390_CRC32_VX)
     if (PREFIX(s390_cpu_has_vx))
-        functable.crc32 = &PREFIX(s390_crc32_vx);
+        ft.crc32 = &PREFIX(s390_crc32_vx);
 #elif defined(X86_PCLMULQDQ_CRC)
     if (x86_cpu_has_pclmulqdq)
-        functable.crc32 = &crc32_pclmulqdq;
+        ft.crc32 = &crc32_pclmulqdq;
 #endif
 
-    return functable.crc32(crc, buf, len);
-}
-
-Z_INTERNAL uint32_t compare256_stub(const uint8_t *src0, const uint8_t *src1) {
-
+    // compare256_stub:
 #ifdef UNALIGNED_OK
 #  if defined(UNALIGNED64_OK) && defined(HAVE_BUILTIN_CTZLL)
-    functable.compare256 = &compare256_unaligned_64;
+    ft.compare256 = &compare256_unaligned_64;
 #  elif defined(HAVE_BUILTIN_CTZ)
-    functable.compare256 = &compare256_unaligned_32;
+    ft.compare256 = &compare256_unaligned_32;
 #  else
-    functable.compare256 = &compare256_unaligned_16;
+    ft.compare256 = &compare256_unaligned_16;
 #  endif
 #else
-    functable.compare256 = &compare256_c;
+    ft.compare256 = &compare256_c;
 #endif
 #if defined(X86_SSE2) && defined(HAVE_BUILTIN_CTZ)
     if (x86_cpu_has_sse2)
-        functable.compare256 = &compare256_sse2;
+        ft.compare256 = &compare256_sse2;
 #endif
 #if defined(X86_AVX2) && defined(HAVE_BUILTIN_CTZ)
     if (x86_cpu_has_avx2)
-        functable.compare256 = &compare256_avx2;
+        ft.compare256 = &compare256_avx2;
 #endif
 #ifdef POWER9
     if (power_cpu_has_arch_3_00)
-        functable.compare256 = &compare256_power9;
-#endif
+        ft.compare256 = &compare256_power9;
+#endif
+
+    functable.adler32 = ft.adler32;
```

Comment:
Add comment above saying, something to the effect of: `// Assign function pointers individually for atomic operation`